### ISSUE TITLE
Extend usertty() to support custom formatting via template(), like file() or program()

### ISSUE THREAD

**Issue by dkl** [syslog-ng/syslog-ng#4066]
## Description of the problem
Currently the `usertty()` destination uses a hard-coded message format. It can be annoying because it's different than the format used by other destinations like `file()` or `program()` and cannot be customized.

## Proposed solution
Maybe `usertty()` could also allow the `template()` option, and apply it when writing to the TTY(s)?

## Alternatives
It's possible to patch syslog-ng to change the hard-coded message format for `usertty()`, but that's bad - it should be configurable via the config file.
Also it's possible to use `program()`, applying the custom `template()` in the process, and pipe the data to an external program which does basically what `usertty()` does, but that's an extra step that could be avoided.

## Additional context
I tried to look into this a bit, but didn't work on this any further:
* It looks like `log_writer_format_log()` is needed to do the formatting. afuser does not use `LogWriter`/`LogPipe` yet at all, but it would probably make sense to convert it to use this infrastructure - if possible. It might have to create one `LogWriter` per TTY found in utmp.
* It might require an alternative to `LogPipe`, something like `LogTty`, to get the proper error handling for `write()` calls (it looks like `LogPipe` does something different than `usertty()` currently)
* Maybe it would be better to refactor `log_writer_format_log()` to allow formatting without a full `LogWriter` (something like a log formatter), then `usertty()` could build the string once and send it to all TTYs, instead of using a separate `LogWriter` instance for each one.
* `LogWriter` might offer other features besides the `template()` formatting that might be interesting for `usertty()` too, I'm not sure.

**Comment by K0001rch** [comment]
i think, that u may simply customize your message, which will be send between remote sessions in `afuser` module.

P.S.:
I tryed to allow using template config with usertty (together), but I couldn't do it =(
The point is that the `template` plugin is presented very implicitly. I did not find the `template` plugin/module representation in the syslog-ng sources.

Instead, I simply customized the messages sent via `usertty` between remote sessions through my own configuration elements by editing the corresponding BISON file

**Comment by bazsi** [comment]
I don't think it would make much sense to convert afuser to use LogWriter, since I wouldn't expect high traffic to be written to the tty. And as long as the traffic is limited the current solution should be good enough.

Adding template based formatting is not difficult, one just needs a new option for the afuser driver added to the grammar, and then calling log_template_format() before the message is emitted.
